pkg/volume/aws_ebs/attacher.go

```diff
@@ -0,0 +1,256 @@
+	return &awsElasticBlockStoreAttacher{host: plugin.host}, nil
+}
+
+func (plugin *awsElasticBlockStorePlugin) GetDeviceName(spec *volume.Spec) (string, error) {
+	volumeSource, _ := getVolumeSource(spec)
+	if volumeSource == nil {
+		return "", fmt.Errorf("Spec does not reference an EBS volume type")
+	}
+
+	return volumeSource.VolumeID, nil
+}
+
+func (attacher *awsElasticBlockStoreAttacher) Attach(spec *volume.Spec, hostName string) error {
+	volumeSource, readOnly := getVolumeSource(spec)
+	volumeID := volumeSource.VolumeID
+
+	awsCloud, err := getCloudProvider(attacher.host.GetCloudProvider())
+	if err != nil {
+		return err
+	}
+
+	attached, err := awsCloud.DiskIsAttached(volumeID, hostName)
+	if err != nil {
+		// Log error and continue with attach
+		glog.Errorf(
+			"Error checking if volume (%q) is already attached to current node (%q). Will continue and try attach anyway. err=%v",
+			volumeID, hostName, err)
+	}
+
+	if err == nil && attached {
+		// Volume is already attached to node.
+		glog.Infof("Attach operation is successful. volume %q is already attached to node %q.", volumeID, hostName)
+		return nil
+	}
+
+	if _, err = awsCloud.AttachDisk(volumeID, hostName, readOnly); err != nil {
+		glog.Errorf("Error attaching volume %q: %+v", volumeID, err)
+		return err
+	}
+	return nil
+}
+
+func (attacher *awsElasticBlockStoreAttacher) WaitForAttach(spec *volume.Spec, timeout time.Duration) (string, error) {
+	awsCloud, err := getCloudProvider(attacher.host.GetCloudProvider())
+	if err != nil {
+		return "", err
+	}
+	volumeSource, _ := getVolumeSource(spec)
+	volumeID := volumeSource.VolumeID
+	partition := ""
+	if volumeSource.Partition != 0 {
+		partition = strconv.Itoa(int(volumeSource.Partition))
+	}
+
+	devicePath := ""
+	if d, err := awsCloud.GetDiskPath(volumeID); err == nil {
+		devicePath = d
+	} else {
+		glog.Errorf("GetDiskPath %q gets error %v", volumeID, err)
+	}
+
+	ticker := time.NewTicker(checkSleepDuration)
+	defer ticker.Stop()
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			glog.V(5).Infof("Checking AWS Volume %q is attached.", volumeID)
+			if devicePath == "" {
+				if d, err := awsCloud.GetDiskPath(volumeID); err == nil {
+					devicePath = d
+				} else {
+					glog.Errorf("GetDiskPath %q gets error %v", volumeID, err)
+				}
+			}
+			if devicePath != "" {
+				devicePaths := getDiskByIdPaths(partition, devicePath)
+				path, err := verifyDevicePath(devicePaths)
+				if err != nil {
+					// Log error, if any, and continue checking periodically. See issue #11321
+					glog.Errorf("Error verifying AWS Volume (%q) is attached: %v", volumeID, err)
+				} else if path != "" {
+					// A device path has successfully been created for the PD
+					glog.Infof("Successfully found attached AWS Volume %q.", volumeID)
+					return path, nil
+				}
+			} else {
+				glog.Errorf("AWS Volume (%q) is not attached yet", volumeID)
```

Comment:
This is not an error. If you want to log it make it `glog.V(5).Infof` otherwise the logs will be filled with this.